internal/pkg/cli/selector/selector.go

```diff
@@ -69,13 +91,110 @@ func NewConfigSelect(prompt Prompter, store *config.Store) *ConfigSelect {
+func NewWorkspaceSelect(prompt Prompter, store AppEnvLister, ws WsSvcLister) *WorkspaceSelect {
 	return &WorkspaceSelect{
 		Select:    NewSelect(prompt, store),
 		svcLister: ws,
 	}
 }
 
+// NewDeploySelect returns a new selector that chooses services and environments from the deploy store.
+func NewDeploySelect(prompt Prompter, configStore AppEnvLister, deployStore DeployStoreClient) *DeploySelect {
+	return &DeploySelect{
+		Select:         NewSelect(prompt, configStore),
+		deployStoreSvc: deployStore,
+	}
+}
+
+// GetServiceEnvironmentOpts sets up optional parameters for GetServiceEnvironmentOpts function.
+type GetServiceEnvironmentOpts func(*DeploySelect)
+
+// WithSvc sets up the svc name for DeploySelect.
+func WithSvc(svc string) GetServiceEnvironmentOpts {
+	return func(in *DeploySelect) {
+		in.svc = svc
+	}
+}
+
+// WithEnv sets up the env name for DeploySelect.
+func WithEnv(env string) GetServiceEnvironmentOpts {
+	return func(in *DeploySelect) {
+		in.env = env
+	}
+}
+
+// ServiceEnvironment fetches all deployed services and environments in an application and then
+// prompts the user to select one combination.
+func (s *DeploySelect) ServiceEnvironment(prompt, help string, app string, opts ...GetServiceEnvironmentOpts) (svc string, env string, err error) {
+	type svcEnv struct {
+		svcName string
+		envName string
+	}
+	for _, opt := range opts {
+		opt(s)
+	}
+	var envNames []string
+	if s.env != "" {
+		envNames = append(envNames, s.env)
+	} else {
+		envNames, err = s.retrieveEnvironments(app)
+		if err != nil {
+			return "", "", fmt.Errorf("list environments: %w", err)
+		}
+	}
+	svcEnvs := make(map[string]svcEnv)
+	var svcEnvNames []string
+	for _, envName := range envNames {
+		var svcNames []string
+		if s.svc != "" {
+			deployed, err := s.deployStoreSvc.IsDeployed(app, envName, s.svc)
+			if err != nil {
+				return "", "", fmt.Errorf("check if service %s is deployed in environment %s: %w", s.svc, envName, err)
+			}
+			if !deployed {
+				continue
+			}
+			svcNames = append(svcNames, s.svc)
+		} else {
+			svcNames, err = s.deployStoreSvc.ListDeployedServices(app, envName)
+			if err != nil {
+				return "", "", fmt.Errorf("list deployed service for environment %s: %w", envName, err)
+			}
+		}
+		for _, svcName := range svcNames {
+			svcEnv := svcEnv{
+				svcName: svcName,
+				envName: envName,
+			}
+			svcEnvName := fmt.Sprintf("%s (%s)", svcName, envName)
+			svcEnvs[svcEnvName] = svcEnv
+			svcEnvNames = append(svcEnvNames, svcEnvName)
+		}
+	}
+	if len(svcEnvNames) == 0 {
+		return "", "", fmt.Errorf("no deployed services found in application %s", color.HighlightUserInput(app))
+	}
+	// return if only one deployed service found
+	if len(svcEnvNames) == 1 {
+		svc = svcEnvs[svcEnvNames[0]].svcName
+		env = svcEnvs[svcEnvNames[0]].envName
+		log.Infof("Showing logs of service %s deployed in environment %s\n", color.HighlightUserInput(svc), color.HighlightUserInput(env))
```

Comment:
Well we can re-use this for things besides logs, right? status?  In fact, if I run copilot svc status, it asks me to choose which logs I want to look at. 

Perhaps we should say `only found one deployed service %s in environment %s`.